apollo-router-core/src/services/http_compat.rs

```diff
@@ -1,78 +1,25 @@
-//! wrapper typpes for Request and Response from the http crate to improve their usability
+//! Wrapper types for [`http::Request`] and [`http::Response`] from the http crate.
+//!
+//! To improve their usability.
 
 use std::{
     cmp::PartialEq,
     hash::Hash,
     ops::{Deref, DerefMut},
-    str::FromStr,
 };
 
-use http::{
-    header::HeaderName, request::Builder, uri::InvalidUri, HeaderMap, HeaderValue, Method, Uri,
-    Version,
-};
+use http::{header::HeaderName, request::Builder, uri::InvalidUri, HeaderValue, Version};
 
 #[derive(Debug)]
 pub struct Request<T> {
     // The goal of having a copy of the url is to keep the right type for `ReqwestSubgraphService` and avoid re-parsing.
     // This url will stay the same than the uri in inner because we only can set a new url with `set_url` method
-    pub(super) url: Uri,
-    inner: http::Request<T>,
+    // pub(super) url: Uri,
+    pub inner: http::Request<T>,
 }
 
+// Most of the required functionality is provided by our Deref and DerefMut implementations.
 impl<T> Request<T> {
-    /// Update the associated URL
```

Comment:
The reason I made this, and also why inner is a private element is because `http::Request` implement `Default` which IMHO makes no sense, we had SOOOO many bugs due to this, because it creates a `Request` with an empty URL which also makes no sense. Using the pattern with a private inner http request and public methods forced us to not directly convert an http::Request into http_compat::Request and then we are sure and confident about the content of this request. It enforce to have a correct `Request` for our usage